### PR TITLE
TT-9741 Add containerSecurityContext

### PR DESCRIPTION
**Pull request by zalbiraw**
## Description

Currently containerSecurityContext is not configurable. This is required as k8s and openshift versions require some of these values to be set.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [ ] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: components/tyk-dashboard/values.yaml [hunk 1]
```diff
@@ -183,6 +183,9 @@ dashboard:
   securityContext:
     runAsUser: 1000
     fsGroup: 2000
+  containerSecurityContext:
+    runAsUser: 999
+    runAsGroup: 999
```
Comment (zalbiraw) [inline components/tyk-dashboard/values.yaml:188]: Why are those set to a different value than the pod securityContext? How come we are not setting the rest of the values

Comment (zalbiraw) [reply components/tyk-dashboard/values.yaml:188]: This is what we set on the rest of the containers 

```
          runAsNonRoot: true
          allowPrivilegeEscalation: false
          privileged: false
          readOnlyRootFilesystem: true
          capabilities:
            drop:
              - all
```

File: tyk-single-dc/values.yaml [hunk 2]
```diff
@@ -559,6 +579,9 @@ tyk-dashboard:
     securityContext:
       runAsUser: 1000
       fsGroup: 2000
+    containerSecurityContext:
+      runAsUser: 999
+      runAsGroup: 999
```
Comment (zalbiraw) [inline tyk-single-dc/values.yaml:584]: If we decide to fix the above. This needs to be changed as well.

File: components/tyk-dashboard/values.yaml (hunk 1 above)
Comment (komalsukhani) [reply components/tyk-dashboard/values.yaml:188]: Sorry I don't get it. Haven't you set introduced those values in this PR?

File: tyk-single-dc/values.yaml [hunk 3]
```diff
@@ -663,6 +686,14 @@ tyk-enterprise-portal:
   securityContext:
     runAsUser: 1000
     fsGroup: 2000
+  containerSecurityContext:
```
Comment (komalsukhani) [inline tyk-single-dc/values.yaml:689]: Can you add a comment above this too like you did with others?

File: components/tyk-enterprise-portal/values.yaml [hunk 4]
```diff
@@ -90,6 +90,14 @@ resources: {}
 securityContext:
   runAsUser: 1000
   fsGroup: 2000
+containerSecurityContext:
```
Comment (komalsukhani) [inline components/tyk-enterprise-portal/values.yaml:93]: Can you add a comment above field to explain what it does.(Even though name is self explanatory!)

**Review by komalsukhani — CHANGES_REQUESTED** [review]

File: components/tyk-dashboard/templates/deployment-dashboard.yaml [hunk 5]
```diff
@@ -56,9 +56,10 @@ spec:
       - image: "{{ .Values.dashboard.image.repository }}:{{ .Values.dashboard.image.tag }}"
         imagePullPolicy: {{ .Values.dashboard.image.pullPolicy }}
         name: dashboard-{{ .Chart.Name }}
+        {{- if .Values.dashboard.containerSecurityContext }}
         securityContext:
-          runAsUser: 999
-          runAsGroup: 999
+        {{- toYaml .Values.dashboard.containerSecurityContext | nindent 12 }}
+        {{- end }}
```
Comment (zalbiraw) [inline components/tyk-dashboard/templates/deployment-dashboard.yaml:62]: @komalsukhani I only copy pasted those out. That's why I asked the question.

Comment (komalsukhani) [reply components/tyk-dashboard/templates/deployment-dashboard.yaml:62]: Sorry my bad. I didn't notice it. I think we should fix it.

File: tyk-single-dc/values.yaml (hunk 3 above)
Comment (komalsukhani) [reply tyk-single-dc/values.yaml:689]: Yeah. That we need to.
I thought we can start with your change. But we can handle it in separate PR.

**Comment by komalsukhani** [comment]
@zalbiraw I also wanted to know how to test this PR. Justing validating securityContext values on deployment or we need to test it on openshift?

**Comment by zalbiraw** [comment]
@komalsukhani yes verifying the values in the deployment is sufficient enough.

File: components/tyk-dashboard/templates/deployment-dashboard.yaml (hunk 5 above)
Comment (zalbiraw) [reply components/tyk-dashboard/templates/deployment-dashboard.yaml:62]: No worries :)

**Review by komalsukhani — APPROVED** [review]

**Pull request merged**